rosbag2/src/rosbag2/rosbag2_node.cpp

```diff
@@ -51,11 +56,83 @@ std::shared_ptr<GenericSubscription> Rosbag2Node::create_generic_subscription(
 
     get_node_topics_interface()->add_subscription(subscription, nullptr);
   } catch (const std::runtime_error & ex) {
-    RCUTILS_LOG_ERROR_NAMED(
-      "rosbag2", "Error subscribing to topic %s. Error: %s", topic.c_str(), ex.what());
+    ROSBAG2_LOG_ERROR_STREAM("Error subscribing to topic '" << topic << "'. Error: " << ex.what());
   }
 
   return subscription;
 }
+std::map<std::string, std::string> Rosbag2Node::get_topics_with_types(
+  const std::vector<std::string> & topic_names)
+{
+  std::vector<std::string> sanitized_topic_names;
+  for (const auto & topic_name : topic_names) {
+    sanitized_topic_names.push_back(topic_name[0] != '/' ? "/" + topic_name : topic_name);
+  }
+
+  // TODO(Martin-Idel-SI): This is a short sleep to allow the node some time to discover the topic
+  // This should be replaced by an auto-discovery system in the future
+  std::this_thread::sleep_for(std::chrono::milliseconds(100));
+  auto topics_and_types = this->get_topic_names_and_types();
+
+  std::map<std::string, std::vector<std::string>> filtered_topics_and_types;
+  for (const auto & topic_and_type : topics_and_types) {
+    if (std::find(sanitized_topic_names.begin(), sanitized_topic_names.end(),
+      topic_and_type.first) != sanitized_topic_names.end())
+    {
+      filtered_topics_and_types.insert(topic_and_type);
+    }
+  }
+
+  return reduce_multiple_types_to_one(filter_topics_with_wrong_types(filtered_topics_and_types));
```

Comment:
do you really need to do this? This copies the complete map. Maybe we could use the first element of the type just by convention.

Reply:
Since this is done at startup time, it's not really a big overhead. Nevertheless, if you think it's ok to just take the first argument by default, that's of course fine with me!